src/editor/Core/Filesystem.re

```diff
@@ -302,6 +302,11 @@ let createConfigIfNecessary = (configDir, file) =>
       )
   );
 
+let getExtensionsFolder = () =>
+  getHomeDirectory()
+  >>= getOniDirectory
+  >>= (dir => getPath(dir, "extensions"));
+
 let getOrCreateConfigFile = filename =>
```

Comment:
Looks like we need an equivalent of this now, `getOrCreateConfigFolder`.

Reply:
I've found myself wanting this now, to add some random languages that don't make much sense to add to the main source here.

This is what I ended up using:

```rust
let getOrCreateConfigFolder = configDir =>
  stat(configDir)
  >>= (
    fun
    | Some(dirStats) =>
      /*
        Check the stats, if its a folder we can stop.
        If its not, we need to make a folder.
       */
      isDir(dirStats)
      /\/= (_ => mkdir(configDir, ()))
      >>= (() => return(configDir))
    | None =>
      /*
       No folder was present, lets make one.
       */
      mkdir(configDir, ()) >>= (() => return(configDir))
  );

let getExtensionsFolder = () =>
  getHomeDirectory()
  >>= getOniDirectory
  >>= (dir => getPath(dir, "extensions"))
  >>= getOrCreateConfigFolder;
```

Reply:
Thanks @CrossR ! Brought this in 👍